### PR TITLE
Design fixes

### PR DESCRIPTION
**Pull request by DanielHeath**

### CODE REVIEW

**Comment by danni** [comment]
You've lost the padding/line-height of the Other contact details link.

File: src/components/Address.js [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
 import ScreenReader from "./ScreenReader";
+import GoogleMapsLink from "./GoogleMapsLink";
+import Location from "../iss/Location";
 
 class Address extends React.Component {
 
     // flow:disable not supported yet
     static sampleProps = {
-        complex: {
+        complex: {location: new Location({
```
Comment (danni) [inline src/components/Address.js]: newline

File: src/components/Address.js [hunk 2]
```diff
@@ -20,60 +20,31 @@ class Address extends React.Component {
...
     render(): ReactElement {
-        var streetParts = [
-            this.props.flat_unit,
-            this.props.level,
-            this.props.building,
-        ].map(text => text.trim());
-
-        streetParts = _(streetParts)
-            .compact()
-            .map(part => `${part}, `)
```
Comment (danni) [inline src/components/Address.js]: This is just `.join(', ')`

Comment (DanielHeath) [inline src/components/Address.js]: It adds it to the last one as well, and doesn't add it if there are none.

File: src/iss/Location.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        var addrDescriptors = [
+            this.props.flat_unit,
+            this.props.level,
+            this.props.building,
+        ].map(text => text.trim());
+
+        addrDescriptors = _(addrDescriptors)
+            .compact()
+            .map(part => `${part}, `)
+            .join("");
```
Comment (danni) [inline src/iss/Location.js:32]: `.join(', ')`

File: src/pages/ResultsListPage.js [hunk 4]
```diff
@@ -149,7 +149,7 @@ class ResultsListPage extends BaseResultsPage {
                         object={object}
                       />;
 
-                    var klass = elem.displayName || "other";
+                    var klass = elem.type.displayName || "other";
```
Comment (danni) [inline src/pages/ResultsListPage.js:152]: ?

File: src/iss/Location.js (hunk 3 above)
Comment (DanielHeath) [inline src/iss/Location.js:32]: That wouldn't be the same thing.

File: src/pages/ResultsListPage.js (hunk 4 above)
Comment (DanielHeath) [inline src/pages/ResultsListPage.js:152]: `elem.displayName` never worked - it always just went to `"other"`, both in dev and prod.

**Comment by danni** [comment]
This is probably going to collide with @pfctdayelise who I think was handling showing the phrase 'Confidential location' if appropriate. Can you make sure that won't break/coordinate a merge.

File: src/iss/Location.js (hunk 3 above)
Comment (DanielHeath) [inline src/iss/Location.js:32]: Specifically, `[]` should result in `""` and `['Level 1']` should result in `"Level 1 "`

File: src/components/ServicePane.js [hunk 5]
```diff
@@ -58,9 +60,16 @@ export default class ServicePane extends React.Component {
 
                     <CollapsedOpeningTimes object={object.open} />
                     <hr />
-                    <Address {...object.location} />
+                    <Address location={object.Location()} />
                     <hr />
                     <ContactMethods object={object} />
+                    <hr />
+                    <GoogleMapsLink
```
Comment (pfctdayelise) [inline src/components/ServicePane.js:67]: Note in mine I was linking to the Google Maps directions (starting point "Current Location")

Comment (DanielHeath) [inline src/components/ServicePane.js:67]: ooh, good spot

File: src/components/TransportTime.js [hunk 6]
```diff
@@ -9,39 +9,70 @@ import icons from "../icons";
...
+        }
+        return <span />;
+    }
+
+    renderPublic(): ReactElement {
+        return (
+            <div
+                className={`TransportTime ${this.compactClass}`}
+            >
+                <icons.Walk className="ColoredIcon" />
```
Comment (pfctdayelise) [inline src/components/TransportTime.js:68]: lost class brand-text-dark? (not that I am sure it does anything for icons...)

File: src/components/TransportTime.scss [hunk 7]
```diff
@@ -1,18 +1,24 @@
...
         position: absolute;
-        top: -5px;
+        top: 5px;
         left: -2px;
     }
 
-    .time {
+    .travel-time {
         color: $brand-text-dark;
         font-weight: bold;
```
Comment (pfctdayelise) [inline src/components/TransportTime.scss]: I asked Danni, this should actually be 500

File: src/components/TransportTime.js (hunk 6 above)
Comment (DanielHeath) [inline src/components/TransportTime.js:68]: Right - I realized that the class wasn't defined in our CSS. Should be a separate commit though.

**Comment by pfctdayelise** [comment]
In terms of the transport time info it looks to me this covers the same things that my change did. :+1:

File: src/iss/Location.js (hunk 3 above)
Comment (danni) [inline src/iss/Location.js:32]: If you want a comma between `addrDescriptors` and `street` just finish with `[a, b].compact().join(', ')`

Comment (DanielHeath) [inline src/iss/Location.js:32]: Not sure that's cleaner - I'll have a look for a good alternative.

File: src/components/Address.js [hunk 8]
```diff
@@ -1,79 +1,62 @@
...
             <div className="Address">
                 <ScreenReader>
                     <h4>Address</h4>
                 </ScreenReader>
-                <a
-                    target="_blank"
-                    aria-label="Open Google Maps in a new tab"
-                    href={`https://maps.google.com/?q=${query}`}
-                >
+                <GoogleMapsLink to={location}>
```
Comment (danni) [inline src/components/Address.js:47]: I thought the Google Maps directions link was meant to go on the `TransportTime` component? Will accept a FIXME comment for this.

File: src/components/GoogleMapsLink.js [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+            children: (
+                <div>Link text</div>
+            ),
+            to: new Location(fixtures.ixa.location),
+        },
+    };
+
+    googleMapsUrl(): string {
+        var toAddr = this.props.to;
+        var start = "Current+Location";
```
Comment (danni) [inline src/components/GoogleMapsLink.js:22]: Would be better as a 'from' property with a default of 'Current Location'.

File: src/pages/ResultsListPage.js (hunk 4 above)
Comment (danni) [inline src/pages/ResultsListPage.js:152]: Okay so this is canonically correct now?

File: src/iss/Location.js (hunk 3 above)
Comment (danni) [inline src/iss/Location.js:32]: I would also accept `join(', ') + ', '`

**Comment by DanielHeath** [comment]
Merging so Isaac can look at it.
